### PR TITLE
op-mode: T3725: Show configuration in JSON format

### PR DESCRIPTION
**Pull request by sever-sever**
## Change Summary

Ability to get configuration from op-mode in JSON format
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)

* https://phabricator.vyos.net/T3725

## Component(s) name

op-mode
## How to test
```
show configuration json
show configuration json pretty
```

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [ ] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [x] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

**Review by c-po — APPROVED** [review]

**Comment by jestabro** [comment]
Strictly speaking, JSON should use double quotes, so perhaps you want json.dumps before output, though I think this is fine just for pretty-printing.

File: src/op_mode/show_configuration_json.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+parser = argparse.ArgumentParser()
+parser.add_argument("-p", "--pretty", action="store_true", help="Show pretty configuration in JSON format")
+
+
+if __name__ == '__main__':
+    args = parser.parse_args()
+
+    if args.pretty:
+        pprint(c)
```
Comment (dmbaturin) [inline src/op_mode/show_configuration_json.py]: As @jestabro rightfully says, JSON export **must** always be done with `json.dumps` to produce syntactically correct JSON. Pretty-print from Python may look _superficially_ like JSON, but isn't.

**Review by dmbaturin — CHANGES_REQUESTED** [review]

**Review by jestabro — APPROVED** [review]

File: src/op_mode/show_configuration_json.py (hunk 1 above)
Comment (dmbaturin) [reply src/op_mode/show_configuration_json.py]: Also, `json.dump` to stdout is faster than `print(json.dumps(c))`.

**Comment by sever-sever** [comment]
> Strictly speaking, JSON should use double quotes, so perhaps you want json.dumps before output, though I think this is fine just for pretty-printing.

Thanks.

File: src/op_mode/show_configuration_json.py (hunk 1 above)
Comment (sever-sever) [reply src/op_mode/show_configuration_json.py]: Sure, I removed pprint and use `indent=4` and use `json.dumps`
Thanks

**Pull request merged**